src/Activity.js

```diff
@@ -1,27 +1,43 @@
 class Activity {
-  constructor(activityData) {
-    this.activityData = activityData
+  constructor(activityData, date, user, userRepo) {
+    this.activityData = activityData;
+    this.userStepsByDate = {};
+    this.date = date;
+    this.user = user;
+    this.userRepo = userRepo;
   }
-  getMilesFromStepsByDate(id, date, userRepo) {
-    let userStepsByDate = this.activityData.find(data => id === data.userID && date === data.date);
-    return parseFloat(((userStepsByDate.numSteps * userRepo.strideLength) / 5280).toFixed(1));
+
+  returnUserStepsByDate() {
+    return this.activityData.find(data => this.user.id === data.userID && this.date === data.date);
+  }
+
+  getMilesFromStepsByDate() {
+    let userStepsByDate = this.returnUserStepsByDate()
+    // let userStepsByDate = this.activityData.find(data => this.user.id === data.userID && this.date === data.date);
+    return parseFloat(((userStepsByDate.numSteps * this.user.strideLength) / 5280).toFixed(1));
   }
-  getActiveMinutesByDate(id, date) {
-    let userActivityByDate = this.activityData.find(data => id === data.userID && date === data.date);
+
+  getActiveMinutesByDate() {
+    let userActivityByDate = this.returnUserStepsByDate();
+    // let userActivityByDate = this.activityData.find(data => this.user.id === data.userID && this.date === data.date);
     return userActivityByDate.minutesActive;
   }
-  calculateActiveAverageForWeek(id, date, userRepo) {
-    return parseFloat((userRepo.getWeekFromDate(date, id, this.activityData).reduce((acc, elem) => {
+
+  calculateActiveAverageForWeek() {
+    const totalActiveMinutes = this.userRepo.getWeekFromDate(this.date, this.user.id, this.activityData)
+    return parseFloat((totalActiveMinutes.reduce((acc, elem) => {
       return acc += elem.minutesActive;
```

Comment:
We can consider adding a parent class to inherit methods such as calculate average. Will DRY up code a bit.